common/src/main/java/com/microsoft/hydralab/common/management/device/TestDevice.java

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package com.microsoft.hydralab.common.management.device;
+
+import com.android.ddmlib.InstallException;
+import com.android.ddmlib.TimeoutException;
+import com.microsoft.hydralab.common.entity.common.DeviceInfo;
+import com.microsoft.hydralab.common.entity.common.TestRun;
+import com.microsoft.hydralab.common.entity.common.TestTask;
+import com.microsoft.hydralab.common.logger.LogCollector;
+import com.microsoft.hydralab.common.screen.ScreenRecorder;
+import com.microsoft.hydralab.common.util.ThreadPoolUtil;
+import com.microsoft.hydralab.common.util.ThreadUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+
+public class TestDevice implements TestDeviceCapability {
```

Comment:
If the TestSession no longer contains the operation, SingleDeviceSession is preferred for me.
Session implies that the lifecycle is the same as TestTask.

SingleDeviceSession, MultiDeviceSession.